src/engine/blkstore/blkstore.hpp

```diff
@@ -336,20 +336,19 @@ class BlkStore {
     BlkAllocStatus alloc_blk(const uint32_t size, blk_alloc_hints& hints, std::vector< BlkId >& out_blkid) {
         // Allocate a block from the device manager
         assert(size % m_pagesz == 0);
-        blk_count_t nblks{static_cast< blk_count_t >(size / m_pagesz)};
-        if (nblks <= BlkId::max_blks_in_op()) {
-            return (m_vdev.alloc_blk(nblks, hints, out_blkid));
-        } else {
-            while (nblks != 0) {
-                static thread_local std::vector< BlkId > result_blkid{};
-                result_blkid.clear();
-                const blk_count_t nblks_op{std::min(static_cast< blk_count_t >(BlkId::max_blks_in_op()), nblks)};
-                const auto ret{m_vdev.alloc_blk(nblks_op, hints, result_blkid)};
-                if (ret != BlkAllocStatus::SUCCESS) { return ret; }
-                out_blkid.insert(std::end(out_blkid), std::make_move_iterator(std::begin(result_blkid)),
-                                 std::make_move_iterator(std::end(result_blkid)));
-                nblks -= nblks_op;
-            }
+        // Using a 16-bit value (blk_count_t) for page counts can lead to overflow issues when managing large storage
+        // devices. With a page size of 4096 bytes, a 16-bit value limits the addressable storage to 256 MB (65,536
+        // pages * 4096 bytes/page).
+        uint32_t nblks{static_cast< uint32_t >(size / m_pagesz)};
```

Comment:
Sorry, I had a second thought on this, I think now the reason for line 340~342 is a shortcut for performance consideration, because 99.99% of the blk_allocation will fall into if, not else. cc @szmyd , can we change it back? sorry.

Reply:
done

Reply:
This looks okay to me; in the future maybe we should move things like this into a subroutine that can more easily be tested with a UT that does not need to go through the alloc_blk call? Just a thought when working in 4.x for situations such as this.